src/test/java/com/fittracker/fittracker/service/WeightServiceTest.java

```diff
@@ -185,5 +193,60 @@ void givenNoDateFound_shouldThrowException() {
             verifyNoMoreInteractions(weightRepository);
         }
     }
+    @Nested
+    class FindByDateRange {
+        private final LocalDate endDate = TEST_DATE.plusDays(3);
+        private final Weight endWeight = new Weight(endDate,TEST_VALUE,TEST_UUID);
+        @Test
+        void givenStartDateAfterEndDate_shouldThrowInvalidDateRangeException() {
+            assertThatThrownBy(()->weightService.findByDateRange(LocalDate.of(2023,5,1),LocalDate.of(2023,1,1)))
+                    .isInstanceOf(InvalidDateRangeException.class)
+                    .hasMessageContaining("Start date cannot be after end date");
+
+            verifyNoInteractions(weightRepository);
+        }
+
+        @Test
+        void givenValidDateRange_shouldReturnListOfWeightResponses() {
+            when(weightRepository.findByDateBetweenAndUserId(TEST_DATE, endDate,TEST_UUID))
+                    .thenReturn(List.of(weight,endWeight));
+
+            var expected = Stream.of(weight,endWeight).map(WeightResponse::fromWeight).toList();
+            var result = weightService.findByDateRange(TEST_DATE, endDate);
+
+            assertThat(result).isEqualTo(expected);
+            verify(weightRepository).findByDateBetweenAndUserId(TEST_DATE, endDate,TEST_UUID);
+            verifyNoMoreInteractions(weightRepository);
+        }
+
+        @Test
+        void givenValidDateRangeWithNoWeights_shouldReturnEmptyList() {
+            when(weightRepository.findByDateBetweenAndUserId(any(),any(),eq(TEST_UUID)))
+                    .thenReturn(List.of());
+
+            var expected = Collections.emptyList();
+            var result = weightService.findByDateRange(TEST_DATE, endDate);
+
+            assertThat(result).isEqualTo(expected);
+            verify(weightRepository).findByDateBetweenAndUserId(TEST_DATE, endDate,TEST_UUID);
+
+            verifyNoMoreInteractions(weightRepository);
+        }
+
+        @Test
+        void givenValidDateRangeWithOneWeight_shouldReturnListWithOneWeightResponse() {
```

Comment:
This is exactly the same scenario as in `givenValidDateRange_shouldReturnListOfWeightResponses`, we get a list from `weightRepository.findByDateBetweenAndUserId` and map it to the list of weight responses.

There's no way for us to check (on unit test level) if the repository does what it should, we're making an assumption that it works correctly here, and check it on integration test level (which we've got covered). So let's remove this test.